TestFailer/TestFailer/include/weatherReportStub.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+#ifdef TESTFAILER_EXPORTS
+#define TESTFAILER_API __declspec(dllexport)
+#else
+#define TESTFAILER_API __declspec(dllimport)
+#endif
+#include <string>
+#include "IWeatherReport.h"
+
+
+namespace WeatherSpace {
+
+    /// This is a stub for a weather sensor. For the sake of testing
+    /// we create a stub that generates weather data and allows us to
+    /// test the other parts of this application in isolation
+    /// without needing the actual Sensor during development
+
+    class TESTFAILER_API SensorStub : public IWeatherSensor {
+
+    public:
+        int Humidity() const override {
+            return 72;
+        }
+
+        int Precipitation() const override {
+            return 70;
+        }
+
+        double TemperatureInC() const override {
+            return 26;
+        }
+
+        int WindSpeedKMPH() const override {
+            return 52;
+        }
+    };
+
+    class TESTFAILER_API FaultySensorStub : public IWeatherSensor {
```

Comment:
Another option is to have one stub with the values passed in the constructor, which it will return when called.
This way, you can have different instances for different weather situations, instead of one class per situation.